arch/posix/CMakeLists.txt

```diff
@@ -1,12 +1,13 @@
 zephyr_compile_options(
-  -fno-freestanding
   -m32
   -MMD
   -MP
   ${ARCH_FLAG}
   -include ${ZEPHYR_BASE}/arch/posix/include/posix_cheats.h
   )
 
+toolchain_cc_freestanding_options()
```

Comment:
Intents need to be documented at the invocation.

Note that when I remarked that this should be a cc option, I actually meant that you could/should write

zephyr_cc_option(-fno-freestanding)

but this works as well.